Cannot compile

I'm trying to compile ddserver on a Ubuntu box. After installing libusb-1.0-0-dev I get the following error.

$ g++ -Wall main.cpp communicator.cpp -I/usr/include/libusb-1.0/ -L/usr/lib /usr/lib/x86_64-linux-gnu/libusb-1.0.a -lpthread -lrt -lstdc++ -o ddserver
In file included from communicator.cpp:8:0:
communicator.h: In constructor ‘Communicator::Communicator(int)’:
communicator.h:46:24: warning: ‘Communicator::mHandle’ will be initialized after [-Wreorder]
  libusb_device_handle *mHandle;
                        ^
communicator.h:45:17: warning:   ‘libusb_device* Communicator::mDevice’ [-Wreorder]
  libusb_device *mDevice;
                 ^
communicator.cpp:10:1: warning:   when initialized here [-Wreorder]
 Communicator::Communicator(int socket) : mSocket(socket), mIsInitialized(false), mIsUsbInitialized(false),
 ^
communicator.cpp: In member function ‘bool Communicator::readFromClient()’:
communicator.cpp:56:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (r == (packetSize - 4)) {
                           ^
communicator.cpp: In member function ‘uint8_t* Communicator::readUsbPacket(int&)’:
communicator.cpp:180:43: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (packetSize1 > (currentPacketSize - 4)) {
                                           ^
communicator.cpp: In member function ‘void Communicator::sendDeviceList(int, std::list<ImagingUsbDevice>*)’:
communicator.cpp:430:45: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘ssize_t {aka long int}’ [-Wformat=]
  syslog(LOG_INFO, "Bytes writen: %d", writen);
                                             ^
communicator.cpp: In member function ‘std::list<ImagingUsbDevice> Communicator::enumerateUsbImagingDevices()’:
communicator.cpp:452:73: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘std::list<ImagingUsbDevice>::size_type {aka long unsigned int}’ [-Wformat=]
   syslog(LOG_INFO, "Imaging USB devices found: %d", imgUsbDevices.size());
                                                                         ^
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "udev_hotplug_event":
(.text+0x1a): riferimento non definito a "udev_device_get_action"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "udev_hotplug_event":
(.text+0x3d): riferimento non definito a "udev_device_get_devnode"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "udev_hotplug_event":
(.text+0x51): riferimento non definito a "udev_device_get_sysname"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "udev_hotplug_event":
(.text+0x7f): riferimento non definito a "udev_device_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_event_thread_main":
(.text+0x202): riferimento non definito a "udev_monitor_receive_device"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x240): riferimento non definito a "udev_new"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x25d): riferimento non definito a "udev_monitor_new_from_netlink"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x27c): riferimento non definito a "udev_monitor_filter_add_match_subsystem_devtype"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x295): riferimento non definito a "udev_monitor_enable_receiving"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x2ab): riferimento non definito a "udev_monitor_get_fd"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x340): riferimento non definito a "udev_monitor_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_start_event_monitor":
(.text+0x366): riferimento non definito a "udev_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_stop_event_monitor":
(.text+0x4b1): riferimento non definito a "udev_monitor_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_stop_event_monitor":
(.text+0x4d2): riferimento non definito a "udev_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5a2): riferimento non definito a "udev_enumerate_new"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5bb): riferimento non definito a "udev_enumerate_add_match_subsystem"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5c3): riferimento non definito a "udev_enumerate_scan_devices"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5cb): riferimento non definito a "udev_enumerate_get_list_entry"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5e4): riferimento non definito a "udev_list_entry_get_name"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x5fd): riferimento non definito a "udev_device_new_from_syspath"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x608): riferimento non definito a "udev_device_get_devnode"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x618): riferimento non definito a "udev_device_get_sysname"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x65b): riferimento non definito a "udev_device_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x663): riferimento non definito a "udev_list_entry_get_next"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_scan_devices":
(.text+0x677): riferimento non definito a "udev_enumerate_unref"
/usr/lib/x86_64-linux-gnu/libusb-1.0.a(libusb_1_0_la-linux_udev.o): nella funzione "linux_udev_hotplug_poll":
(.text+0x6e8): riferimento non definito a "udev_monitor_receive_device"
collect2: error: ld returned 1 exit status



Do I miss some other package? I'm not a coder so sorry for the silly question

--- a/README.md
+++ b/README.md
@@ -29,38 +29,18 @@ DslrDashboardServer
  
  Under **Multimedia** select **ddserver**. Build OpenWrt or just a single package. The resulting package can be found in the *bin/[platform]/packages/ddserver_0.1-1.ipk* directory. If you selected * the package will be already installed in the OpenWrt image.
  Upon installation DDserver will add an init script so you can start/stop it from the web interface (System->Startup).
- 
-## Compile DslrDashboard Server on Linux
- 
- Switch to the *DslrDashboardServer/src* directory, and compile the source code:
- 
- 	g++ -Wall main.cpp communicator.cpp -I/usr/include/libusb-1.0/ -L/usr/lib /usr/lib/i386-linux-gnu/libusb-1.0.a lpthread -lrt -lstdc++ -o ddserver
 
-or on x64 systems
 
-	g++ -Wall main.cpp communicator.cpp -I/usr/include/libusb-1.0/ -L/usr/lib /usr/lib/x86_64-linux-gnu/libusb-1.0.a lpthread -lrt -lstdc++ -o ddserver
+## Compile DslrDashboard Server on Debian based distributions, such as Ubuntu and Raspbian
+
+	sudo apt-get install build-essential pkg-config libusb-1.0-0-dev
+	g++ -Wall src/main.cpp src/communicator.cpp `pkg-config --libs --cflags libusb-1.0` -lpthread -lrt -lstdc++ -o ddserver
+
+## Compile DslrDashboard Server on other Linux distributions
+
+	g++ -Wall src/main.cpp src/communicator.cpp -I/usr/include/libusb-1.0/ -lusb-1.0 -lpthread -lrt -lstdc++ -o ddserver
 
 Make the resulting *ddserver* binary executable and launch the server:
 
 	chmod +x ddserver
 	./ddserver
-
-## Compile DslrDashboard Server on Raspberry Pi
-
-Install Git and libusb-1.0-0-dev:
- 
- 	sudo apt-get install git libusb-1.0-0-dev
- 
- Clone the *DslrDashboardServer* repository and switch to the *src* directory:
- 
- 	git clone git://github.com/hubaiz/DslrDashboardServer
- 	cd DslrDashboardServer/src
- 
-## Compile the server using the command bellow:
- 
- 	g++ -Wall main.cpp communicator.cpp -I/usr/include/libusb-1.0/ -L/usr/lib /usr/lib/arm-linux-gnueabihf/libusb-1.0.a -lpthread -lrt -lstdc++ -o ddserver
- 
- Make the resulting *ddserver* binary executable and launch the server:
- 
- 	chmod +x ddserver
- 	./ddserver

